server/src/matchmaking/matchmaking.gateway.ts

```
@@ -45,9 +47,16 @@ export class MatchmakingGateway implements OnGatewayDisconnect {
     @MessageBody('difficulty') diff: Difficulty,
     @MessageBody('auth') token,
   ) {
-    // Call some method from matchmaking service here
     try {
       const user = this.jwtService.verify(token);
+      if (isEmpty(diff))
+        throw new BadRequestException('"difficulty" is a required parameter');
+
+      const isValidDifficulty: boolean =
+        Object.values(Difficulty).includes(diff);
+
+      if (!isValidDifficulty)
+        throw new BadRequestException(`${diff} is not a valid difficulty`);
```

I don't think you can throw this exception here, since they're meant for REST endpoints, not gateways